tests/unit/test_shorthand.py

```diff
@@ -184,6 +202,61 @@ def test_parse(data, expected):
     actual = shorthand.ShorthandParser().parse(data)
     assert actual == expected
 
+class TestShorthandParserParamFile:
+    @pytest.fixture()
+    def files(self):
+        files = FileCreator()
+        yield files
+        files.remove_all()
+
+    @pytest.mark.parametrize(
```

Comment:
I was thinking we could generalize the following tests:
* `test_paramfile`
* `test_file_assignment_no_file`
* `test_file_param_without_file_assignment`

by parameterizing the file contents, the input to the `parse` method, and the expected result.